File: JavaScript/batch_renamer_25-feb-23.js
/******** 
 * Author :- Sandip Nandi
 * Date of Creation - 26-Feb-23
 *  Address :- Bnagalore, Karnataka
 *  Batch Renamer
 *
*********/



const fs = require('fs');
const path = require('path');

const type_filter = '.txt';
const search_File = 'file';
const replace_File = 'document';

// filter the current directory for files
const dir = fs.readdirSync('.');
const files = dir.filter(f => fs.statSync(f).isFile());

console.log(`${files.length} of ${dir.length} elements are files.`);

let renamed = 0;

for (const doc of files) {
  const file = path.parse(doc);

  // skip not matching file types
  if (file.ext != type_filter) continue;
  // skip not matching file names
  if (!file.name.includes(search_File)) continue;

  // rename the actual file
  fs.renameSync(doc, doc.replace(search_File, replace_File));
  renamed++;
}

console.log(`renamed ${renamed} of ${files.length} files.`);

File: Python/batch_renamer_25-feb-23.py
########################################################
#  Author :- Sandip Nandi                              #
#  Creation Date :- 25-Feb-23                          #
#  Address :- Bengaluru,Karnataka                      #
#  Purpose :- Want to rename Files, execute it         #
#  Batch Renamer                                       #
########################################################

import os

## to be replaced string and file extension filter

searchFile = "document"

replaceFile = "file"

type_filter = ".py"

## get all files from current directory

dir_contents = os.listdir('.')

docs = [doc for doc in dir_contents if os.path.isfile(doc)]
renamed = 0

print(f"{len(docs)} of {len(dir_contents)} elements are files.")

## go through all the files and check if they match the search pattern
for doc in docs:
    ## separate name from file extension
    doc_name, filetype = os.path.splitext(doc)

    ## filter for files with the right extension
    if filetype == type_filter:
        ## check if search text is in doc name
        if searchFile in doc_name:
            
            ## replace with the given text
            new_name = doc_name.replace(searchFile, replaceFile) + filetype
            os.rename(doc, new_name)
            renamed += 1

            print(f"Renamed file {doc} to {new_name}")

print(f"Renamed {renamed} of {len(docs)} files.")



